File: Examples/VisualBasic/Text/ExtractingTextFromAPage.vb
Imports System.IO
Imports Aspose.Note
Imports System.Collections.Generic

Namespace Text
    Public Class ExtractingTextFromAPage
        Public Shared Sub Run()
            ' ExStart:ExtractingTextFromAPage
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Text()

            ' Load the document into Aspose.Note.
            Dim oneFile As New Document(dataDir & Convert.ToString("Aspose.one"))

            ' Get list of page nodes
            Dim nodes As IList(Of Node) = oneFile.GetChildNodes(Of Node)()

            If nodes.Count > 0 AndAlso nodes(0).NodeType = NodeType.Page Then
                Dim page As Page = DirectCast(nodes(0), Page)
                ' Retrieve text
                Dim text As String = page.GetText()
                ' Print text on the output screen
                Console.WriteLine(text)
            End If
            ' ExEnd:ExtractingTextFromAPage
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Text/ExtractingTextFromAPage.cs

using System.IO;
using Aspose.Note;
using System;
using System.Collections.Generic;
using System.Linq;

namespace Aspose.Note.Examples.CSharp.Text
{
    public class ExtractingTextFromAPage
    {
        public static void Run()
        {
            // ExStart:ExtractingTextFromAPage
            // ExFor:Page.GetChildNodes
            // ExFor:RichText
            // ExFor:RichText.Text
            // ExSummary:Shows how to get all text from the page.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Text();

            // Load the document into Aspose.Note.
            Document oneFile = new Document(dataDir + "Aspose.one");

            // Get list of page nodes
            var page = oneFile.GetChildNodes<Page>().FirstOrDefault();

            if (page != null)
            {
                // Retrieve text
                string text = string.Join(Environment.NewLine, page.GetChildNodes<RichText>().Select(e => e.Text)) + Environment.NewLine;
                // Print text on the output screen
                Console.WriteLine(text);
            }

            // ExEnd:ExtractingTextFromAPage
        }
    }
}